src/server/req_rescq.c

```diff
@@ -222,6 +222,199 @@ cnvrt_timer_init()
  */
 
 
+/**
+ * @brief
+ * 		remove_node_from_resv - procedure removes node from reservation
+ *		the node is removed from RESV_ATR_resv_nodes and assigned
+ *		resources are accounted back to loaner's pool and finally the
+ *		reservation is removed from the node
+ *
+ * @parm[in,out]	presv	-	reservation structure
+ * @parm[in,out]	pnode	-	pointer to node
+ *
+ */
+void
+remove_node_from_resv(resc_resv *presv, struct pbsnode *pnode) {
+	char *begin = NULL;
+	char *end = NULL;
+	char *tmp_buf;
+	struct	resvinfo *rinfp, *prev;
+	attribute   tmpatr;
+
+	tmp_buf = (char *)malloc(strlen(pnode->nd_name) + 2);
+	if (tmp_buf == NULL) {
+		snprintf(log_buffer, LOG_BUF_SIZE-1, "malloc failure (errno %d)", errno);
+			log_err(PBSE_SYSTEM, __func__, log_buffer);
+		return;
+	}
+
+	strcpy(tmp_buf, pnode->nd_name);
+	tmp_buf[strlen(pnode->nd_name)] = ':';
+	tmp_buf[strlen(pnode->nd_name) + 1] = '\0';
```

Comment:
we can use a single sprintf() call instead of the above 3 statements.

Reply:
OK, done.